game/system/hid/display_manager.cpp

```diff
@@ -300,6 +300,46 @@ void DisplayManager::set_display_mode(game_settings::DisplaySettings::DisplayMod
   }
 }
 
+void DisplayManager::toggle_display_mode() {
+  const auto current_mode = m_display_settings.display_mode;
+  // Store the current prefered fullscreen mode
+  if (current_mode != game_settings::DisplaySettings::DisplayMode::Windowed) {
+    m_previous_fullscreen_display_mode = current_mode;
+  }
+  lg::info("Current display mode: ");
+  switch (current_mode) {
+    case game_settings::DisplaySettings::DisplayMode::Fullscreen:
+      lg::info("Fullscreen\n");
+      lg::info("Switching to Windowed mode...\n");
+      enqueue_set_window_display_mode(game_settings::DisplaySettings::DisplayMode::Windowed);
+      break;
```

Comment:
You can combine the fullscreen and borderless cases as they do the same thing:
```suggestion
    case game_settings::DisplaySettings::DisplayMode::Fullscreen:
    case game_settings::DisplaySettings::DisplayMode::Borderless:
      lg::info("Fullscreen/Borderless\n");
      lg::info("Switching to Windowed mode...\n");
      enqueue_set_window_display_mode(game_settings::DisplaySettings::DisplayMode::Windowed);
      break;
```

Reply:
Oh thats actually really cool, I was thinking of converting it to an IF statement instead, this is much better thanks! Noted.